p8e-shared/src/main/kotlin/io/provenance/p8e/shared/util/SmartKeyAuthRefresh.kt

```diff
@@ -0,0 +1,29 @@
+package io.provenance.p8e.shared.util
+
+import com.fortanix.sdkms.v1.api.AuthenticationApi
+import io.provenance.p8e.shared.extension.logger
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+
+@Component
+class SmartKeyAuthRefresh(
+    private val authenticationApi: AuthenticationApi
+) {
+
+    private val log = logger()
+
+    companion object {
+        const val REFRESH_RATE: Long = 300_000 // refresh at a rate of every 5 minutes.
+    }
+
+    /**
+     * Refresh the Auth Token every 5 Minutes
```

Comment:
Does refresh use the existing api key to fetch a new one or only based off of the api secret key?

Reply:
It extends the current session. According to Smart Key:

`Refreshes an existing session, thus extending the expiration time of the existing bearer authentication token`

Ill reword the comment